src/main/java/com/artipie/http/SliceByPath.java

```diff
@@ -76,18 +76,13 @@ public Response response(final String line, final Iterable<Map.Entry<String, Str
         final Publisher<ByteBuffer> body) {
         final Key key;
         try {
-            final String[] split = new RequestLineFrom(line).uri().getPath()
-                .replaceAll("^/+", "").split("/");
-            if (this.settings.layout().equals("org")) {
-                if (split.length < 2) {
-                    throw new IllegalStateException("Expected at least 2 path segments");
-                }
-                key = new Key.From(split[0], split[1]);
-            } else {
-                if (split.length < 1) {
-                    throw new IllegalStateException("Expected at least 1 path segment");
-                }
-                key = new Key.From(split[0]);
+            key = this.keyFromPath(new RequestLineFrom(line).uri().getPath());
+            if (key.equals(Key.ROOT)) {
```

Comment:
@genryxy I see that you use `Key.ROOT` as special value that means repository name cannot be extracted from path. That is not that great design to have such values with special handling. It will be more obvious to return `Optional<Key>` from method and have special handling for absent value